core/src/HIP/Kokkos_HIP_Space.cpp

```diff
@@ -195,6 +204,61 @@ void* HIPHostPinnedSpace::impl_allocate(
 
   return ptr;
 }
+
+#if defined KOKKOS_ENABLE_HIP_UNIFIED_MEMORY
+void* HIPUnifiedSpace::allocate(const size_t arg_alloc_size) const {
+  return allocate("[unlabeled]", arg_alloc_size);
+}
+void* HIPUnifiedSpace::allocate(const char* arg_label,
+                                const size_t arg_alloc_size,
+                                const size_t arg_logical_size) const {
+  return impl_allocate(arg_label, arg_alloc_size, arg_logical_size);
+}
+void* HIPUnifiedSpace::impl_allocate(
+    const char* arg_label, const size_t arg_alloc_size,
+    const size_t arg_logical_size,
+    const Kokkos::Tools::SpaceHandle arg_handle) const {
+  void* ptr = nullptr;
+
+  if (arg_alloc_size > 0) {
+    if (Kokkos::Impl::num_hip_um_allocations.load() == 0) {
+      const char* hsa_xnack = std::getenv("HSA_XNACK");
+      if (Kokkos::show_warnings() and not(hsa_xnack))
```

Comment:
```suggestion
      if (Kokkos::show_warnings() and !hsa_xnack)
```
I don't think we use that syntax anywhere else.